ebcl/tools/secure_aptrepo/secure_repo.py

```diff
@@ -0,0 +1,96 @@
+import base64
+import argparse
+from functools import partial
+import os
+import signal
+import sys
+import threading
+from http.server import SimpleHTTPRequestHandler, HTTPServer
+
+
+class AuthHTTPRequestHandler(SimpleHTTPRequestHandler):
+    def __init__(self, *args, username, password, directory=None, **kwargs):
+        self.username = username
+        self.password = password
+        if directory is None:
+            directory = os.getcwd()
+        super().__init__(*args, directory=directory, **kwargs)
+
+    def do_AUTHHEAD(self):
+        self.send_response(401)
+        self.send_header('WWW-Authenticate', 'Basic realm="Secure Area"')
+        self.send_header('Content-type', 'text/html')
+        self.end_headers()
+
+    def do_GET(self):
+        auth_header = self.headers.get('Authorization')
+        if auth_header is None:
+            self.do_AUTHHEAD()
+            self.wfile.write(b'No auth header received')
+        elif self.authenticate(auth_header):
+            SimpleHTTPRequestHandler.do_GET(self)
+        else:
+            self.do_AUTHHEAD()
+            self.wfile.write(b'Invalid credentials')
+
+    def authenticate(self, auth_header):
+        encoded_credentials = auth_header.split(' ')[1]
+        decoded_credentials = base64.b64decode(encoded_credentials).decode('utf-8')
+        username, password = decoded_credentials.split(':')
+        return username == self.username and password == self.password
+
+
+class Server:
+    def __init__(self, port, username, password, directory):
+        self.port = port
+        self.username = username
+        self.password = password
+        self.directory = directory
+        self.httpd = None
+        self.server_thread = None
+
+    def start(self):
+        handler = partial(
+            AuthHTTPRequestHandler,
+            username=self.username,
+            password=self.password,
+            directory=self.directory
+        )
+        self.httpd = HTTPServer(('', self.port), handler)
+        print(f'Serving APT repository from {self.directory} on https://localhost:{self.port}')
```

Comment:
```suggestion
        print(f'Serving APT repository from {self.directory} on https://localhost:{self.port}')
```
I guess this is only http, not https?

Reply:
accepted and same will be updated